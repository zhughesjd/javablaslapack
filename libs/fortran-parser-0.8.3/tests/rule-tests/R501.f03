! R501, intrinsic_type_spec, no attrs, w/ double colon
integer :: x
integer y
integer, pointer :: xptr
integer, pointer, allocatable :: yptr

! R501, test entity_decl_list
real :: a,b

! R501, intrinsic_type_spec with kind_selector
! kind_selector is R404.
integer(kind=4) :: m

! R501, intrinsic_type_spec with char_selector
! char_selector is R424.
character(kind=1, len=10) :: string

end
