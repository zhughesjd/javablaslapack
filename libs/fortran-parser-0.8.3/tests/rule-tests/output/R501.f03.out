*******************************************
args[1]: R501.f03
R1101:main-program__begin:
R 403:intrinsic-type-spec: keyword1=[@2,55:61='integer',<61>,2:0] keyword2=null type=400 hasKindSelector=false
R 502:declaration-type-spec: type=500
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@6,66:66='x',<209>,2:11]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 403:intrinsic-type-spec: keyword1=[@8,68:74='integer',<61>,3:0] keyword2=null type=400 hasKindSelector=false
R 502:declaration-type-spec: type=500
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@10,76:76='y',<209>,3:8]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 403:intrinsic-type-spec: keyword1=[@12,78:84='integer',<61>,4:0] keyword2=null type=400 hasKindSelector=false
R 502:declaration-type-spec: type=500
R 503:attr-spec: attr=814
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@19,98:101='xptr',<209>,4:20]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=1
R 403:intrinsic-type-spec: keyword1=[@21,103:109='integer',<61>,5:0] keyword2=null type=400 hasKindSelector=false
R 502:declaration-type-spec: type=500
R 503:attr-spec: attr=814
R 503:attr-spec: attr=805
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@31,136:139='yptr',<209>,5:33]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=2
R 403:intrinsic-type-spec: keyword1=[@36,172:175='real',<62>,8:0] keyword2=null type=401 hasKindSelector=false
R 502:declaration-type-spec: type=500
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@40,180:180='a',<209>,8:8]
R 504:entity-decl: id=[@42,182:182='b',<209>,8:10]
R 504:entity-decl-list: count=2
R 501:type-declaration-stmt: numAttributes=0
R 406:int-literal-constant: digitString=[@53,270:270='4',<10>,12:13] kindParam=null
R 404:kind-selector: token1=[@51,265:268='kind',<188>,12:8] token2=[@52,269:269='=',<26>,12:12] hasExpression=true
R 403:intrinsic-type-spec: keyword1=[@49,257:263='integer',<61>,12:0] keyword2=null type=400 hasKindSelector=true
R 502:declaration-type-spec: type=500
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@58,276:276='m',<209>,12:19]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 406:int-literal-constant: digitString=[@69,366:366='1',<10>,16:15] kindParam=null
R 402:type-param-value: hasExpr=true hasAsterisk=false hasColon=false
R 424:char-selector: kindOrLen1=301 kindOrLen2=300 hasAsterisk=false
R 406:int-literal-constant: digitString=[@74,373:374='10',<10>,16:22] kindParam=null
R 402:type-param-value: hasExpr=true hasAsterisk=false hasColon=false
R 424:char-selector: kindOrLen1=301 kindOrLen2=302 hasAsterisk=false
R 403:intrinsic-type-spec: keyword1=[@65,351:359='character',<64>,16:0] keyword2=null type=405 hasKindSelector=true
R 502:declaration-type-spec: type=500
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@79,380:385='string',<209>,16:29]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 204:specification-part: numUseStmts=0 numImportStmts=0 numDeclConstructs=7
R1103:end-program-stmt: id=null
R1101:main-program: hasProgramStmt=false hasExecutionPart=false hasInternalSubprogramPart=false
Parser exiting normally
********************************************
